training.api/Controllers/CidadeController.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using training.api.Model;
+
+namespace training.api.Controllers
+{
+    [ApiController]
+    [Route("[controller]")]
+    public class CidadeController : ControllerBase
+    {
+
+        private readonly TrainingContext context;
+
+        public CidadeController(TrainingContext context)
+        {
+            this.context = context;
+        }
+
+        [HttpGet]
+        public ActionResult<IEnumerable<Cidade>> GetAll()
+        {
+            var cidades = context.Cidades;
+            return Ok(cidades);
+        }
+
+        [HttpPost]
+        public ActionResult<Cidade> AddCidade(string estado, string nome)
+        {
+            Estado? est = context.Estados.Where(e => e.UF == estado).FirstOrDefault();
+            Cidade cid = context.Cidades.Where(c => c.Nome == nome && c.IdEstado == est.Id).FirstOrDefault();
+            if (cid == null)
+            {
+                if (est == null)
+                {
+                    Estado newEstado = new Estado();
+
+                    newEstado.UF = estado;
+
+                    context.Estados.Add(newEstado);
+                    context.SaveChanges();
+                    est = newEstado;
+                    return Ok(newEstado);
+                }
+                Cidade newCidade = new Cidade();
+
+                newCidade.IdEstado = est.Id;
+                newCidade.Nome = nome;
+
+                context.Cidades.Add(newCidade);
+                context.SaveChanges();
+                return Ok(newCidade);
+            }
+            else
+            {
+                return Problem();
```

Comment:
```
    return BadRequest("A cidade já existe e não pode ser criada");
```

Se o retorno não for auto explicativo e.g. `NotFound` tem que colocar mensagem de erro.